src/radchallenge.c

```diff
@@ -0,0 +1,210 @@
+/*
+ * radchallenge.c - a sample c program showing how to embed the configuration of a radius and to challenge
+ * client, using the FreeRADIUS Client Library without an external configuration and dictionary file .
+ */
+
+#include <stdlib.h>
+#include <sys/types.h>
+#include <syslog.h>
+#include <radcli/radcli.h>
+#include <string.h>
+
```

Comment:
Would you like to move it to tests/ and make it run on travis similarly to the existing tests? (e.g., basic-tests). That way we can be sure that this new feature will not regress in the future.

Reply:
I have no idea how to connect tests and config generation without config file.
Should it be as enough flexible as `radiusclient.c`? Or maybe, should it be static?

It would be nice to save challenge example (because there are no other challenge examples here)

Reply:
Check Makefile.am for the existing tests. basic-tests.sh is a good start. The server in all the tests is fixed and its config is at: tests/docker/radius-clients-notcp.conf